sycl/include/sycl/group_algorithm.hpp

```diff
@@ -883,9 +945,33 @@ std::enable_if_t<
     OutPtr>
 joint_exclusive_scan(Group g, InPtr first, InPtr last, OutPtr result,
                      BinaryOperation binary_op) {
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  static_assert(
+      (std::is_same_v<BinaryOperation,
+                      sycl::logical_or<std::remove_cv_t<
+                          std::remove_reference_t<decltype(*first)>>>> ||
```

Comment:
Can you create a temporary type alias to simplify this?

```
  // Maybe use std::decay_t instead?
  using elem_t = std::remove_cv_t<std::remove_reference_t<decltype(*first)>>;
```
  
Same down below